custom_components/ramses_cc/__init__.py

```diff
@@ -264,44 +242,11 @@ def __init__(
     @property
     def available(self) -> bool:
         """Return True if the sensor is available."""
-        return getattr(self._device, self._state_attr) is not None
+        return getattr(self.rf_entity, self._state_attr) is not None
 
     @property
     def name(self) -> str:
         """Return the name of the binary_sensor/sensor."""
-        if not hasattr(self._device, "name") or not self._device.name:
-            return f"{self._device.id} {self._state_attr}"
-        return f"{self._device.name} {self._state_attr}"
-
-
-class RamsesZoneBase(RamsesEntity):  # for: climate & water_heater
```

Comment:
This is almost entirely unused so it seemed most appropriate to just move anything relevant into the individual entity classes as it made things easier to understand.

Reply:
Yep.